lib/rspec/support/object_inspector.rb

```diff
@@ -4,23 +4,42 @@ module Support
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
       # @api private
-      def self.inspect(object)
-        if Time === object
+      def self.inspect(initial)
+        formatted = formatter(initial)
+        if String === formatted || Symbol === formatted
```

Comment:
So I see that the `Symbol === formatted` check is needed to make [this test](https://github.com/rspec/rspec-support/blob/ad758245d098c24b9fd4bd0bb7a4054e66b574c2/spec/rspec/support/object_inspector_spec.rb#L81-L83) pass but I think we should enforce that only strings are treated this way here.  IMO, the test is buggy, as the matcher protocol states that `description` should return a string, not a symbol.  We should update the `description` method.